ledger/participant-integration-api/src/main/scala/platform/store/cache/MutableCacheBackedContractStore.scala

```diff
@@ -115,9 +115,9 @@ private[platform] class MutableCacheBackedContractStore(
   )(implicit
```

Comment:
When I was making changes here it was not necessarily obvious to me that we need the lookup/partitioning for all the cached entries first (only optimizes the algorithm in terms of db lookups in corner cases).
We might would be able to do this sequentially, which could remove some complexity here.